__init__.py

```diff
@@ -0,0 +1,333 @@
+        elif window_units == "lines":
+            regex = r"^(.+)\n+|(.+)\n+|(.+)$"
+            lines = []
+            for match in re.finditer(regex, doc.text):
+                start, end = match.span()
+                span = doc.char_span(start, end)
+                if span is not None:
+                    lines.append(span)
+            return lines
+        else:
+            return doc
+
+    @timer
+    def calculate(
+        self,
+        patterns: Union[list, str] = None,
+        calculator: Union[Callable, str] = "rw_calculator",
+        query: str = "counts",
+        show_spacy_rules: bool = False,
+    ) -> None:
+        """Set up a calculator.
+
+        Args:
+            patterns: (Union[list, str]): The patterns to search for.
+                        calculator (Union[Callable, str]): The calculator to use.
+            query (str): String designating whether to return "counts", "averages", or "ratios".
+                        show_spacy_rules (bool): Whether to use spaCy rules or strings in column labels
+        """
+        if not hasattr(self, "windows"):
+            raise Exception("You must call set_windows() before running calculations.")
+        else:
+            if calculator:
+                # Use the "averages" calculator with the default config
+                if isinstance(calculator, str):
+                    if patterns is not None:
+                        self.patterns = patterns
+                    calculator = get_rw_component(calculator)
+                    calculator = calculator(
+                        patterns=self.patterns, windows=self.windows, query=query
+                    )
+            calculator.run(query=calculator.query)
+            self.metadata["calculator"] = calculator.metadata
+            self.result = calculator.to_df(show_spacy_rules=show_spacy_rules)
+
+    def plot(
+        self,
+        plotter: Union[Callable, str] = "rw_simple_plotter",
+        show: bool = False,
+        file: str = None,
+        **kwargs,
+    ) -> None:
+        """Set up the plotter.
+
+        Args:
+            plotter (Union[Callable, str]): The plotter to use.
+            show (bool): Whether to show the generated figure.
+            file (str): The filepath to save the file, if desired.
+        """
+        if not hasattr(self, "result") or self.result is None:
+            raise Exception(
+                "You must run a calculator on your data before generating a plot."
+            )
+        # Use the "rw_simple_plotter" plotter with the default config
+        if isinstance(plotter, str):
+            plotter = get_rw_component(plotter)
+            plotter = plotter()
+        plotter.run(self.result)
+        self.metadata["plotter"] = plotter.metadata
+        self.fig = plotter.fig
+        if show:
+            plotter.show(**kwargs)
+        if file:
+            plotter.save(file, **kwargs)
+
+    # @timer
+    def set_windows(
```

Comment:
Some validation on `window_units` matching the options would be helpful here. I was able to do this:
```
import rollingwindows
rw = rollingwindows.RollingWindows(doc, model=model)
rw.set_windows("aaa", "aaa")
rw.metadata

{'model': 'en_core_web_sm',
 'n': 'aaa',
 'window_units': 'aaa',
 'alignment_mode': 'strict',
 'search_method': 're_finditer',
```
Running `rw.calculate()` later then failed, but you could handle this upfront when configuring the window options instead of potentially getting a runtime error. I'd validate the parameters here to make sure that you're getting an `int` and a valid string option.